packages/components/src/components/input-field/input-field.jsx

```diff
@@ -184,10 +185,13 @@ const InputField = ({
     };
 
     const updateValue = (new_value, is_long_press) => {
-        const formatted_value = format ? format(new_value) : new_value;
+        let formatted_value = format ? format(new_value) : new_value;
         if (is_long_press) {
             setLocalValue(formatted_value);
         } else {
+            if (is_signed && !/^[+-]/.test(formatted_value) && Math.sign(formatted_value) > 0) {
```

Comment:
```suggestion
            if (is_signed && /^\d+$/.test(formatted_value) && formatted_value > 0) {
```
if you're checking whether the +/- is already added or not, a simple is number check should be suffice
if you're checking for sign, if it is larger than zero it is positive?

Reply:
Will replace the regex